src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_VOLUME = 10;
+    private int size = 0;
+    private Object[] list;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_VOLUME];
+    }
```

Comment:
dont use raw types, when our class is parameterized generic:
```suggestion
    private T[] list;

    public ArrayList() {
        list = (T[]) new Object[DEFAULT_VOLUME];
    }
```